youtube_dl/downloader/external.py

```diff
@@ -182,7 +182,8 @@ class Aria2cFD(ExternalFD):
     AVAILABLE_OPT = '-v'
 
     def _make_cmd(self, tmpfilename, info_dict):
-        cmd = [self.exe, '-c']
+        cmd = [self.exe]
+        cmd += self._valueless_option('-c', 'continuedl')
```

Comment:
This has no effect for youtube-dl's `--continue`/`--no-continue` - in both cases downloading is resumed. This is because you must also pass `--allow-overwrite=true` and `--remove-control-file=true` is case of youtube-dl's `--no-continue`.

Reply:
Use full `--continue=true|false` notation instead.

Reply:
> This has no effect for youtube-dl's `--continue`/`--no-continue` - in both cases downloading is resumed. This is because you must also pass `--allow-overwrite=true` and `--remove-control-file=true` is case of youtube-dl's `--no-continue`.

Do you mean I should check for ydl's `no-overwrites` flag and add aria2's `allow-overwrite` here? I can do it, but isn't it better to change that in a new PR? That logic wasn't there before.

But if you mean to add `allow-overwrite` for `no-continue`, you misunderstand my intention. The scenario I'm having is actually not overwriting file with conflict output name, but instead saving those files with different names thanks to aria2's auto renaming feature. Imagine I have a file "test.mp4" and try to download another "test.mp4". Instead of overwriting or continuing, I want aria2 to download it and rename the new file to "test.1.mp4". If more, "test.2.mp4", "test.3.mp4".

If user wants to overwrite, they can always put the flags you mentioned in their aria2.conf. In my opinion, youtube-dl should pass as few hard coded flags to external downloader as possible, because user can't control them.

> Use full --continue=true|false notation instead.

Will do.

Reply:
>Do you mean I should check for ydl's `no-overwrites` flag and add aria2's `allow-overwrite` here?

No. `--no-overwrites` is for non media files.

>But if you mean to add `allow-overwrite` for `no-continue`, you misunderstand my intention.

The point is not your intention but replicating the actual behavior of youtube-dl itself. With `--continue` youtube-dl resumes download, aria2c as external downloader should do the same, so that a command should be passed to aria2c that will instruct it to resume. When `--no-continue` is used youtube-dl downloads from scratch, same should do aria2c.

Reply:
> When --no-continue is used youtube-dl downloads from scratch, same should do aria2c.

I agree with this. But "from scratch" does not necessarily mean to always overwrite existing (unrelated) file just because they have same name. There is a good reason why aria2 provides `allow-overwrite` and default it to false in the first place.

If youtube-dl hardcode `--allow-overwrite=true` here, user who does not want to overwrite can't change the behavior. If youtube-dl does not hardcode, user who does want to overwrite can still provide that flag in their `--external-downloader-args`. Wouldn't the more flexible approach be better? Or maybe am I missing some project design principle that favors hardcoding commonly used combinations of flags?

Reply:
I've already pointed out: by default it should work the way youtube-dl works. If user wants to override external downloader behavior he should explicitly provide desired options via `--external-downloader-args` (these options should override the defaults by any suitable way external downloader supports - by putting at front/back or whetever).
As for link mentioned that's not a problem at all as it's easily solved by proper output template.

Reply:
Updated. Please take a look.

> these options should override the defaults by any suitable way external downloader supports - by putting at front/back or whetever

I usually try to avoid passing conflict arguments to external programs because the order of override is generally implementation detail. For some first appeared wins while for others the last appeared wins (this is the case for aria2). They could change the behavior from version to version since it is not a contract.